service/src/main/java/com/zebrunner/reporting/service/util/GitHubClient.java

```diff
@@ -0,0 +1,141 @@
+package com.zebrunner.reporting.service.util;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.zebrunner.reporting.domain.db.ScmAccount;
+import com.zebrunner.reporting.domain.dto.scm.ScmConfig;
+import lombok.Setter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PreDestroy;
+import java.io.IOException;
+
+@Component
+@Slf4j
```

Comment:
General notes: avoid using `this.` outside constructors and setters. Too many empty lines in methods